### ISSUE TITLE
test

### ISSUE THREAD

**Issue by jaehyunkim0211** [jaehyunkim0211/flask#1]
Function CalculateVacation(rng As Range) As Double
    Dim cell As Range
    Dim total As Double
    total = 0

    For Each cell In rng
        ' 셀이 날짜인지 또는 문자열인지 확인
        If IsDate(cell.Value) Or IsText(cell.Value) Then
            If cell.Interior.Color = RGB(255, 255, 255) Then ' 흰색(반차가 아닌 경우)
                total = total + 1
            ElseIf cell.Interior.Color <> RGB(255, 255, 255) Then ' 흰색이 아닌 경우(반차인 경우)
                total = total + 0.5
            End If
        End If
    Next cell

    CalculateVacation = total
End Function

Function IsText(value As Variant) As Boolean
    IsText = VarType(value) = vbString
End Function

**Comment by jaehyunkim0211** [comment]
import pandas as pd
import numpy as np
from sklearn.preprocessing import MinMaxScaler, OneHotEncoder

def load_and_preprocess_files(file_paths):
    """
    파일 경로 리스트를 받아 각 파일을 읽어들이고, 전처리한 데이터프레임을 반환하는 함수.
    
    :param file_paths: CSV 파일 경로 리스트
    :return: 전처리된 데이터프레임들의 리스트
    """
    scaler = MinMaxScaler()
    encoder = None
    processed_dfs = []

    for file_path in file_paths:
        df = pd.read_csv(file_path)
        df[['sensor_value', 'order']] = scaler.fit_transform(df[['sensor_value', 'order']])
        
        if encoder is None:
            df, encoder = encode_categorical_data(df)
        else:
            df, _ = encode_categorical_data(df, encoder)
        
        processed_dfs.append(df)
    
    return processed_dfs

def encode_categorical_data(df, encoder=None):
    """
    sensor_name과 item을 원핫인코딩하는 함수
    
    :param df: 데이터프레임
    :param encoder: 기존 인코더 (없으면 None)
    :return: 인코딩된 데이터프레임과 인코더 객체
    """
    if encoder is None:
        encoder = OneHotEncoder(sparse=False)
        encoded_data = encoder.fit_transform(df[['sensor_name', 'item']])
    else:
        encoded_data = encoder.transform(df[['sensor_name', 'item']])
    
    encoded_columns = encoder.get_feature_names_out(['sensor_name', 'item'])
    encoded_df = pd.DataFrame(encoded_data, columns=encoded_columns)
    
    df = df.drop(columns=['sensor_name', 'item'])
    df = pd.concat([df.reset_index(drop=True), encoded_df.reset_index(drop=True)], axis=1)
    
    return df, encoder

def create_sequences_from_files(processed_dfs, seq_length):
    """
    처리된 데이터프레임 리스트를 받아 각 파일별로 시퀀스를 생성하고 RNN 모델 학습에 사용할 데이터를 반환하는 함수
    
    :param processed_dfs: 전처리된 데이터프레임들의 리스트
    :param seq_length: 시퀀스의 길이
    :return: 각 파일의 시퀀스(X)와 타겟(y)의 리스트
    """
    X_list = []
    y_list = []
    
    for df in processed_dfs:
        X, y = create_sequences(df.values, seq_length)
        
        if len(X) > 0:
            X_list.append(X)
            y_list.append(y)
    
    return np.vstack(X_list), np.vstack(y_list)

def create_sequences(data, seq_length):
    """
    주어진 데이터에서 시퀀스를 생성하는 함수
    
    :param data: 넘파이 배열 형태의 데이터
    :param seq_length: 시퀀스 길이
    :return: X, y 시퀀스 배열
    """
    X, y = [], []
    for i in range(len(data) - seq_length):
        X.append(data[i:i + seq_length, :])
        y.append(data[i + seq_length, :])
    
    return np.array(X), np.array(y)

# 파일 경로 리스트
file_paths = ['/mnt/data/sensor1.csv', '/mnt/data/sensor2.csv']  # 실제로는 100개 이상의 파일 경로가 들어감

# 데이터 전처리
processed_dfs = load_and_preprocess_files(file_paths)

# 시퀀스 생성
seq_length = 3
X, y = create_sequences_from_files(processed_dfs, seq_length)

# 결과 확인
print("Final X shape:", X.shape)
print("Final y shape:", y.shape)
